tools/test/test_depsolve.py

```diff
@@ -311,3 +1302,91 @@ def test_depsolve(tmp_path, repo_servers, dnf_config, detect_fn, test_case):
+def test_dump(tmp_path, repo_servers, dnf_config, detect_fn, test_case):
+    try:
+        detect_fn()
+    except RuntimeError as e:
+        pytest.skip(e)
+
+    repo_servers_copy = repo_servers.copy()
+    if "additional_servers" in test_case:
+        repo_servers_copy.extend(test_case["additional_servers"])
+
+    for repo_configs, root_dir, opt_metadata in config_combos(tmp_path, repo_servers_copy):
+        with TemporaryDirectory() as cache_dir:
+            res, exit_code = dump(repo_configs, root_dir, cache_dir, dnf_config, opt_metadata)
+
+            if test_case.get("error", False):
+                assert exit_code != 0
+                assert res["kind"] == test_case["error_kind"]
+                assert re.match(test_case["error_reason_re"], res["reason"], re.DOTALL)
+                continue
+
+            assert exit_code == 0
+            assert len(res) == test_case["packages_count"]
+
+            for res_pkg in res:
+                for key in ["arch", "buildtime", "description", "epoch", "license", "name", "release", "repo_id",
+                            "summary", "url", "version"]:
+                    assert key in res_pkg
+                if res_pkg["name"] == "pkg-with-no-deps":
+                    assert res_pkg["repo_id"] == "custom"
+                else:
+                    assert res_pkg["repo_id"] == "baseos"
+
+            # if opt_metadata includes 'filelists', then each repository 'repodata' must include a file that matches
+            # *filelists*
+            n_filelist_files = len(glob(f"{cache_dir}/*/repodata/*filelists*"))
+            if "filelists" in opt_metadata:
+                assert n_filelist_files == len(REPO_PATHS)
+            else:
+                assert n_filelist_files == 0
+
+
+@pytest.mark.parametrize("test_case", search_test_cases, ids=tcase_idfn)
+@pytest.mark.parametrize("dnf_config, detect_fn", [
+    (None, assert_dnf),
+    ('{"use_dnf5": false}', assert_dnf),
+    ('{"use_dnf5": true}', assert_dnf5),
+], ids=["no-config", "dnf4", "dnf5"])
+def test_search(tmp_path, repo_servers, dnf_config, detect_fn, test_case):
+    try:
+        detect_fn()
+    except RuntimeError as e:
+        pytest.skip(e)
+
+    repo_servers_copy = repo_servers.copy()
+    if "additional_servers" in test_case:
+        repo_servers_copy.extend(test_case["additional_servers"])
+
+    search_args = test_case["search_args"]
+
+    for repo_configs, root_dir, opt_metadata in config_combos(tmp_path, repo_servers_copy):
+        with TemporaryDirectory() as cache_dir:
+            res, exit_code = search(search_args, repo_configs, root_dir, cache_dir, dnf_config, opt_metadata)
+
+            if test_case.get("error", False):
+                assert exit_code != 0
+                assert res["kind"] == test_case["error_kind"]
+                assert re.match(test_case["error_reason_re"], res["reason"], re.DOTALL)
+                continue
+
+            assert exit_code == 0
+            print(res)
```

Comment:
Is this a leftover debug print or here intenionally

Reply:
Yes, thanks!